webapp/channels/src/utils/desktopAPI.ts

```diff
@@ -0,0 +1,288 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import semver from 'semver';
+
+import {isDesktopApp} from 'utils/user_agent';
+
+// TODO: Temporary typings
+import type {DesktopAPI} from '../../../../../desktop/src/types/api/main';
+declare global {
+    interface Window {
+        desktopAPI?: DesktopAPI;
+    }
+}
+
+class DesktopApp {
+    private name?: string;
+    private version?: string | null;
+    private dev?: boolean;
+
+    /**
+     * @deprecated
+     */
+    private postMessageListeners?: Map<string, Set<(message: unknown) => void>>;
+
+    constructor() {
+        // Check the user agent string first
+        if (!isDesktopApp()) {
+            return;
+        }
+
+        this.getDesktopAppInfo().then(({name, version}) => {
+            this.name = name;
+            this.version = semver.valid(semver.coerce(version));
+
+            // Legacy Desktop App version, used by some plugins
+            if (!window.desktop) {
+                window.desktop = {};
+            }
+            window.desktop.version = semver.valid(semver.coerce(version));
+        });
+        window.desktopAPI?.isDev?.().then((isDev) => {
```

Comment:
desktopApp and desktopAPI are two names too similar XD Not sure if it is worth a name change, but I got really confused about how this will work because I thought this was `window.desktopApp`, defined in `plugins/exports.js`.

Reply:
Hehe I know. To be honest, I'd actually prefer to use `window.desktop`, but because of the legacy stuff on there it was confusing...

Open to suggestions on how to improve it :)

Reply:
No good ideas on how to improve it :(